app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/summary/SummaryProductFragment.kt

```diff
@@ -462,6 +477,45 @@ class SummaryProductFragment : BaseFragment(), ISummaryProductPresenter.View {
         }
     }
 
+    private fun showListChips() {
+
+        // remove the existing childviews on chip group if any
+        binding.listChips.removeAllViews()
+
+        val asyncSessionList = OFFApplication.daoSession.startAsyncSession()
+        asyncSessionList.queryList(OFFApplication.daoSession.yourListedProductDao.queryBuilder()
+                .where(YourListedProductDao.Properties.Barcode.eq(product.code)).build())
+
+        val queryResult= arrayListOf<YourListedProduct>()
+        asyncSessionList.listenerMainThread = AsyncOperationListener { operation ->
+            (operation.result as List<YourListedProduct> ).mapTo(queryResult) {it}
+
+            if (!queryResult.isNullOrEmpty()) {
+                queryResult.forEach{ list->
+                    val chip = Chip(context)
+                    chip.text = list.listName
+
+                    // set a random color to the chip's background, we want a dark background as our text color is white so we will limit our rgb to 180
+                    val chipColor: Int = Color.rgb(Random.nextInt(180),Random.nextInt(180),Random.nextInt(180) )
+                    chip.chipBackgroundColor = ColorStateList.valueOf(chipColor)
+                    chip.setTextColor(Color.WHITE)
+
+                    // open list when the user clicks on chip
+                    chip.setOnClickListener {
+                        activity?.startActivity(Intent(activity, ProductListActivity::class.java).apply {
```

Comment:
I think it would be better if we used `ProductListActivity.start`

Reply:
Looking into it.

Reply:
Yup, ```ProductListActivity.start``` is a better chioice. Thanks.